docs/DeveloperGuide.adoc

```diff
@@ -570,10 +677,44 @@ image::list/ListActivityDiagram.png[]
 //=== [Proposed] Undo/Redo feature
 //==== Proposed Implementation
 
-// tag::dataencryption[]
-//=== [Proposed] Data encryption
+//tag::dataencryption[]
+=== [Proposed] Data encryption
+With the emphasis on information security these days, we are concerned about protecting the privacy of our users. We have
+discussed on how we can protect our users' data on `**THRIFT**` by making use of modern encryption schemes to protect against
+prying eyes. The encryption will take place at the back-end with minimal disruptions to the user. The user does not need additional
+knowledge about how the encryption work as `**THRIFT**` will handle all encryption and decryption operations for them.
+
+==== Proposed implementation
+In order to ensure a seamless experience, we propose a new `**Keystore**` module, along with a `**KeystoreManager**` that will
+be interacting closely with the `**Logic**` and `**Storage**` modules. The class diagram below gives an overview of how the
+new `**Keystore**` would fit in:
+
+image::encryption/EncryptionClassDiagram.png[]
 
-//_{Explain here how the data encryption feature will be implemented}_
+We recommend that the `**KeystoreManager**` implements the following methods:
+
+* `KeystoreManager#setEncryptionScheme(Cipher)` - Sets the encryption scheme to be that of `Cipher`.
+* `KeystoreManager#encryptThrift(Thrift)` - Encrypts the current `**Thrift**` object with the encryption scheme set in `**KeystoreManager**`.
+* `KeystoreManager#decryptThrift()` - Decrypts the encrypted text file, restore the `**Thrift**` object and replace the current `**Thrift**`.
+
+TIP: The encryption and decryption will take place every time after `LogicManager#execute()` is called.
```

Comment:
Hmmm... shouldn't decryption takes place in `start()` while encryption takes place when saving data to storage is required.